int32 offset_scale
---
int32 err
